assets/js/main.js

```diff
@@ -1024,173 +992,79 @@ function updatePlatformSwitch() {
-    $('#platform_ps4').removeClass('list-group-item-success');
-    $('#platform_pc').removeClass('list-group-item-success');
-    $('#platform_xb1').addClass('list-group-item-success');
-    Cookies.set('platform', 'xb1');
-    platformSwapped = true;
-    getWorldState();
-    setTimeout(updatePlatformSwitch, 30000);
-  });
+selectPlatform(Cookies.get('platform'));
+
+// Platform switcher
+$('.platform-picker li').click(e => {
+  const platform = $(e.currentTarget).attr('data-platform');
+  selectPlatform(platform);
+  Cookies.set('platform', platform);
+  platformSwapped = true;
+  getWorldState();
+  setTimeout(updatePlatformSwitch, 30000);
 });
 
-$(() => {
-  $('.component-check').click(e => {
-    const status = $(`#${e.target.id}`).prop('checked') ? 'true' : 'false';
-    switch (e.target.id) {
-    case 'earth_checkbox':
-      Cookies.set('earth', status);
-      break;
-    case 'cetus_checkbox':
-      Cookies.set('cetus', status);
-      break;
-    case 'voidtrader_checkbox':
-      Cookies.set('voidtrader', status);
-      break;
-    case 'darvo_checkbox':
-      Cookies.set('darvo', status);
-      break;
-    case 'sales_checkbox':
-      Cookies.set('sales', status);
-      break;
-    case 'acolyte_checkbox':
-      Cookies.set('acolytes', status);
-      break;
-    case 'alerts_checkbox':
-      Cookies.set('alerts', status);
-      break;
-    case 'bounties_checkbox':
-      Cookies.set('bounties', status);
-      break;
-    case 'sortie_checkbox':
-      Cookies.set('sortie', status);
-      break;
-    case 'fissures_checkbox':
-      Cookies.set('fissures', status);
-      break;
-    case 'news_checkbox':
-      Cookies.set('news', status);
-      break;
-    case 'invasions_checkbox':
-      Cookies.set('invasions', status);
-      break;
-    case 'reset_checkbox':
-      Cookies.set('reset', status);
-      updateResetTime();
-      break;
-    default:
-      break;
+// Set default component selections if there aren't any
+[['acolytes'], ['cetus'], ['earth'], ['bounties'], ['alerts'],
+  ['news'], ['invasions'], ['reset'], ['sortie'], ['fissures'],
+  ['baro'], ['darvo'], ['deals', 'false']].forEach(([component, defValue]) => {
+  let value = Cookies.get(component);
+  if (typeof value === 'undefined') {
+    if (typeof defValue === 'undefined') {
+      value = 'true';
+    } else {
+      value = defValue;
     }
-    refreshSelections();
-  });
+    Cookies.set(component, value);
+  }
+  if (value === 'true') {
+    $(`.component-check[data-component="${component}"]`)
+      .prop('checked', true);
+  } else {
+    $(`#component-${component}`).hide();
+  }
 });
 
-// Set default component selections if there aren't any
-function setDefaultCookies() {
-  // Set default platform to PC if there isn't one
-  if (typeof Cookies.get('platform') === 'undefined') {
-    Cookies.set('platform', 'pc');
+// Prevent components menu from closing on label click
+$('#component-selector > ul').click(e => {
```

Comment:
i like the previous way better, but this works fine...

Reply:
This solution makes use of Bootstrap's dropdowns. And it's a bit shorter. I'm pretty sure they use similar code to previous solution but why reinvent the wheel when you can just stop click event propagation. :D